### PR TITLE
fix: Add SCM selector to in-a-box

### PR DESCRIPTION
**Pull request by catto**
I added a feature for supporting bitbucket addressed in #321.

##### Output
```
🎁   Boxing up Screwdriver
👀   Checking prerequisites
🔐   Generating signing secrets
📤   Which SCM provider would you like to use? (github/bitbucket) a
📤   Which SCM provider would you like to use? (github/bitbucket) bitbucket
📦   Generating OAuth credentials

    Please create a new OAuth application on Bitbucket.com
    Go to https://bitbucket.org/account/user/<your username>/oauth-consumers/new to start the process
    For 'URL' put http://192.168.1.16:9000
    For 'Callback URL' put http://192.168.1.16:9001/v4/auth/login
    for 'Permissions' enable Read checkbox for Repositories, Account and Pull requests
    When done, please provide the following values:

    Key: thisiskey
    Secret:

💾   Writing Docker Compose file
🚀   Screwdriver is ready to launch!
(snip)
```
docker-compose.yml (excerpt)
```
        environment:
            PORT: 80
            URI: http://192.168.1.16:9001
            ECOSYSTEM_UI: http://192.168.1.16:9000
            ECOSYSTEM_STORE: http://192.168.1.16:9002
            SCM_PLUGIN: bitbucket
            DATASTORE_PLUGIN: sequelize
            DATASTORE_SEQUELIZE_DIALECT: sqlite
            EXECUTOR_PLUGIN: docker
            SECRET_WHITELIST: "[]"
            EXECUTOR_DOCKER_DOCKER: |
                {
                    "socketPath": "/var/run/docker.sock"
                }
            SECRET_OAUTH_CLIENT_ID: thisiskey
            SECRET_OAUTH_CLIENT_SECRET: thisissecret

### CODE REVIEW

**Comment by catto** [comment]
done!

File: in-a-box.py [hunk 1]
```diff
@@ -140,26 +141,65 @@ def generate_jwt():
...
+    if scm_plugin == 'github':
+        service_name = 'GitHub.com'
+        start_url = 'https://github.com/settings/applications/new'
+        homepage_url = 'homepage url'
+        callback_url = 'authorization callback url'
+        additional_process = ''
+        client_id_name = '    Client ID: '
+        client_secret_name = 'Client Secret'
+    elif scm_plugin == 'bitbucket':
+        service_name = 'Bitbucket.com'
```
Comment (d2lam) [inline in-a-box.py]: Bitbucket.org

File: in-a-box.py [hunk 2]
```diff
@@ -140,26 +141,65 @@ def generate_jwt():
...
+        ip = ip,
+        start_url = start_url,
+        service_name = service_name,
+        homepage_url = homepage_url,
+        callback_url = callback_url,
+        additional_process = additional_process
+    ))
+
+    client_id = get_input('    %s: ' % client_id_name)
+    secret = getpass.getpass('    %s: ' % client_secret_name)
```
Comment (stjohnjohnson) [inline in-a-box.py]: Eh, padding is now all off here.  :(

```bash
    When done, please provide the following values:

        Client ID: : 123456
    Client Secret:
```

File: in-a-box.py [hunk 3]
```diff
@@ -140,26 +141,65 @@ def generate_jwt():
...
     ip: str
         The IP address
     """
+    if scm_plugin == 'github':
+        service_name = 'GitHub.com'
+        start_url = 'https://github.com/settings/applications/new'
+        homepage_url = 'homepage url'
+        callback_url = 'authorization callback url'
+        additional_process = ''
+        client_id_name = '    Client ID: '
```
Comment (stjohnjohnson) [inline in-a-box.py]: Remove the spacing here and the `:`

File: in-a-box.py [hunk 4]
```diff
@@ -140,26 +141,65 @@ def generate_jwt():
...
     ----------
+    scm_plugin: str
+        The SCM provider
     ip: str
         The IP address
     """
+    if scm_plugin == 'github':
+        service_name = 'GitHub.com'
+        start_url = 'https://github.com/settings/applications/new'
+        homepage_url = 'homepage url'
```
Comment (stjohnjohnson) [inline in-a-box.py]: Lower-case here vs Upper-case in bitbucket?

Comment (catto) [inline in-a-box.py]: I copied `URL` from bitbucket. It's better that this is copied from github.

File: in-a-box.py (hunk 2 above)
Comment (catto) [inline in-a-box.py]: Hmm.. I couldn't reproduce this. Is it caused by an indent at L192?

Comment (catto) [inline in-a-box.py]: fixed!

**Pull request merged**